app/dojo.py

```diff
@@ -133,6 +134,73 @@ def allocate_available_livingspace(self, new_person):
 			return (text_format.CRED + "{} has been added to the livingspace waiting list\n" 
 				.format(new_person.name)
 				+ text_format.CEND)
-		
+
+	def print_room(self, room_name):
+		"""This method prints names of all occupants in the
+		specified room name
+
+		"""
+		output = ""
+
+		for room in itertools.chain(self.all_offices, self.all_livingspace) :
+			if room.name == room_name:
+				output = ("\n LIST OF ALL OCCUPANTS IN " + room.room_type +  " " + room_name + "\n" + "*" * 50)
+				if room.occupants:
+					for occupant in room.occupants:
+						output += ("\n" + occupant.name + "\t" + occupant.role + "\n")
+				else:
+					output+= ("\n\n\tThe {} {} has no occupants\n\n".format(room.room_type,room_name))
+			else:
+				output = ("\n\tThe room {} does not exist!\n" .format(room_name))
+			return (text_format.CBOLD + output + text_format.CEND)
+
+
+	def print_allocations(self,filename=None):
+		"""This method prints a list of all allocations at the Dojo. The registered
+		allocations can then be written to the specified text file
+
+		"""
+		output = ""
+
+		for room in itertools.chain(self.all_offices, self.all_livingspace):
+			if room.occupants:
+				output = ("\n\nROOM NAME: {} \tTYPE: {} " .format(room.name, room.room_type))
+				output += ("\n" + "-" * 40 + "\n")
+				for occupant in room.occupants:
+					output += (occupant.name + "-" + occupant.role + ", ")
+
+
+		if filename:
+			print("Saving output data to file...")
+			txt_file = open(filename + ".txt", "w+")
+			txt_file.write(output)
+			txt_file.close()
+			return ("\nData has been successfully saved to {}.txt\n" .format(filename))
+		else:
+			return output
+				
+
+	def print_unallocated(self,filename=None):
+		"""This method prints a list of all staff and fellows,
+		that have not been allocated any office or living space.
+
+		"""
+
+		output = ""
+
+		if self.officespace_waitinglist or self.livingspace_waitinglist:
+			output = "\n\n LIST OF ALL UNALLOCATED STAFF AND FELLOWS\n" + "*" * 50 + "\n"
+			for person in itertools.chain(self.officespace_waitinglist, self.livingspace_waitinglist):
+				output += (person.name + " " + person.email + " " + person.role + " " + person.wants_accomodation)
+
+		if filename:
+			print ("Saving unallocations list to file...")
```

Comment:
There is a better alternative to this by using `with open (filename + ".txt", "w+") as txt_file: txt_file.write(output)`

No need to close the file as it does that itself